src/lib/utils/package-manifest.ts

```diff
@@ -205,8 +206,25 @@ export function getTsEntryPointForPackage(
         typeof packageJson.main == "string"
     ) {
         packageMain = packageJson.main;
+    } else if (
+        hasOwnProperty(packageJson, "types") &&
+        typeof packageJson.types == "string"
+    ) {
+        packageMain = packageJson.types;
```

Comment:
Should we first allow the default value of index.js to be tested for existence before we resort to the types field?
I could imagine a package which had an index.ts, an index.js, and index.d.ts, and therefore doesn't specify package.main, 

With this change typedoc would now parse index.d.ts for docs rather than index.ts, while it's probably preferable to parse index.ts.

Essentially I'm proposing we only check the types field if we hit the condition in line 220 in the old implementation (although there may be another edge case or intent weirdness I'm missing).

Reply:
Yes, I've changed that behaviour while having something else on my mind:

On the other hand, usually when `main` & `types` are specified, it means that dev has specified a separate file created for storing his types definitions (case when your project is written in JS, but you've added TS types only to not rewrite whole project in TS), so logically we should check out `types`/`typings` first. Previously we had `includeDeclarations` flag for controlling whenever you wanted to include `.d.ts` files in documenting process, but AFAIK it's gone since v0.20.

But for now I'll make it follow behaviour specified in README.